File: name-validation-2-dot-net/REST/NV2Response.cs

namespace name_validation_2_dot_net.REST
{
    public class NameInfoV2Response
    {
        public NameInfoV2 NameInfoV2 { get; set; }

        public Error Error { get; set; }
        public override string ToString()
        {
            return $"NameInfoV2: {NameInfoV2}\n" +
                $"Error: {{{Error}}}";
        }
    }
    public class NameInfoV2
    {
        public BestGuessName BestGuessName { get; set; }
        public string NameIn { get; set; }
        public string NameClassification { get; set; }
        public string Prefix { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string Suffix { get; set; }
        public bool FirstNameFound { get; set; }
        public bool IsCommonFirstName { get; set; }
        public string FirstNameOrigin { get; set; }
        public string FirstNameSimilar { get; set; }
        public bool LastNameFound { get; set; }
        public bool IsCommonLastName { get; set; }
        public string LastNameOrigin { get; set; }
        public string LastNameSimilar { get; set; }
        public string Gender { get; set; }
        public string FirstNameAlt { get; set; }
        public string MiddleNameAlt { get; set; }
        public string LastNameAlt { get; set; }
        public bool FirstNameAltFound { get; set; }
        public bool LastNameAltFound { get; set; }
        public string GenderAlt { get; set; }
        public string RelatedNames { get; set; }
        public bool IsCorrectedName { get; set; }
        public bool IsBusinessName { get; set; }
        public string BusinessName { get; set; }
        public int VulgarityScore { get; set; }
        public int CelebrityScore { get; set; }
        public int BogusScore { get; set; }
        public int GarbageScore { get; set; }
        public int FirstNameDictionaryScore { get; set; }
        public int MiddleNameDictionaryScore { get; set; }
        public int LastNameDictionaryScore { get; set; }
        public int OverallNameScore { get; set; }
        public string IsNameGood { get; set; }
        public string StatusCodes { get; set; }
        public string Status { get; set; }
        public override string ToString()
        {
            string Output = $"{{BestGuessName: {BestGuessName}\n" +
                $"NameIn: {NameIn}\n" +
                $"NameClassification: {NameClassification}\n" +
                $"Prefix: {Prefix}\n" +
                $"FirstName:  {FirstName}\n" +
                $"MiddleName: {MiddleName}\n" +
                $"LastName:  {LastName}\n" +
                $"Suffix:   {Suffix}\n" +
                $"FirstNameFound: {FirstNameFound}\n" +
                $"IsCommonFirstName: {IsCommonFirstName}\n" +
                $"FirstNameOrigin: {FirstNameOrigin}\n" +
                $"FirstNameSimilar: {FirstNameSimilar}\n" +
                $"LastNameFound: {LastNameFound}\n" +
                $"IsCommonLastName: {IsCommonLastName}\n" +
                $"LastNameOrigin: {LastNameOrigin}\n" +
                $"LastNameSimilar: {LastNameSimilar}\n" +
                $"Gender: {Gender}\n" +
                $"FirstNameAlt: {FirstNameAlt}\n" +
                $"MiddleNameAlt: {MiddleNameAlt}\n" +
                $"LastNameAlt: {LastNameAlt}\n" +
                $"FirstNameAltFound: {FirstNameAltFound}\n" +
                $"LastNameAltFound: {LastNameAltFound}\n" +
                $"GenderAlt: {GenderAlt}\n" +
                $"RelatedNames: {RelatedNames}\n" +
                $"IsCorrectedName: {IsCorrectedName}\n" +
                $"IsBusinessName: {IsBusinessName}\n" +
                $"BusinessName: {BusinessName}\n" +
                $"VulgarityScore: {VulgarityScore}\n" +
                $"CelebrityScore: {CelebrityScore}\n" +
                $"BogusScore: {BogusScore}\n" +
                $"GarbageScore: {GarbageScore}\n" +
                $"FirstNameDictionaryScore: {FirstNameDictionaryScore}\n" +
                $"MiddleNameDictionaryScore: {MiddleNameDictionaryScore}\n" +
                $"LastNameDictionaryScore: {LastNameDictionaryScore}\n" +
                $"OverallNameScore: {OverallNameScore}\n" +
                $"IsNameGood: {IsNameGood}\n" +
                $"StatusCodes: {StatusCodes}\n" +
                $"Status: {Status}\n";
            return Output;
        }

    }
    public class BestGuessName
    {
        public string Prefix { get; set; }
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string Suffix { get; set; }
        public override string ToString()
        {
            return $"{{Prefix: {Prefix}\n" +
            $"FirstName: {FirstName}\n" +
            $"MiddleName: {MiddleName}\n" +
            $"LastName: {LastName}\n" +
            $"Suffix: {Suffix} }}\n";
        }
    }
    public class Error
    {
        public string Type { get; set; }

        public string TypeCode { get; set; }

        public string Desc { get; set; }

        public string DescCode { get; set; }
        public override string ToString()
        {
            return $"Type: {Type}\n" +
                $"TypeCode: {TypeCode}\n" +
                $"Desc: {Desc}\n" +
                $"DescCode: {DescCode} ";
        }
    }

}

File: name-validation-2-nodejs/REST/nv2_response.js
class BestGuessName {
    constructor(data = {}) {
        this.Prefix = data.Prefix;
        this.FirstName = data.FirstName;
        this.MiddleName = data.MiddleName;
        this.LastName = data.LastName;
        this.Suffix = data.Suffix;
    }

    toString() {
        return `{Prefix: ${this.Prefix}\n` +
               `FirstName: ${this.FirstName}\n` +
               `MiddleName: ${this.MiddleName}\n` +
               `LastName: ${this.LastName}\n` +
               `Suffix: ${this.Suffix}}`;
    }
}

class NameInfoV2 {
    constructor(data = {}) {
        this.BestGuessName = data.BestGuessName ? new BestGuessName(data.BestGuessName) : "";
        this.NameIn = data.NameIn;
        this.NameClassification = data.NameClassification;
        this.Prefix = data.Prefix;
        this.FirstName = data.FirstName;
        this.MiddleName = data.MiddleName;
        this.LastName = data.LastName;
        this.Suffix = data.Suffix;
        this.FirstNameFound = data.FirstNameFound;
        this.IsCommonFirstName = data.IsCommonFirstName;
        this.FirstNameOrigin = data.FirstNameOrigin;
        this.FirstNameSimilar = data.FirstNameSimilar;
        this.LastNameFound = data.LastNameFound;
        this.IsCommonLastName = data.IsCommonLastName;
        this.LastNameOrigin = data.LastNameOrigin;
        this.LastNameSimilar = data.LastNameSimilar;
        this.Gender = data.Gender;
        this.FirstNameAlt = data.FirstNameAlt;
        this.MiddleNameAlt = data.MiddleNameAlt;
        this.LastNameAlt = data.LastNameAlt;
        this.FirstNameAltFound = data.FirstNameAltFound;
        this.LastNameAltFound = data.LastNameAltFound;
        this.GenderAlt = data.GenderAlt;
        this.RelatedNames = data.RelatedNames;
        this.IsCorrectedName = data.IsCorrectedName;
        this.IsBusinessName = data.IsBusinessName;
        this.BusinessName = data.BusinessName;
        this.VulgarityScore = data.VulgarityScore;
        this.CelebrityScore = data.CelebrityScore;
        this.BogusScore = data.BogusScore;
        this.GarbageScore = data.GarbageScore;
        this.FirstNameDictionaryScore = data.FirstNameDictionaryScore;
        this.MiddleNameDictionaryScore = data.MiddleNameDictionaryScore;
        this.LastNameDictionaryScore = data.LastNameDictionaryScore;
        this.OverallNameScore = data.OverallNameScore;
        this.IsNameGood = data.IsNameGood;
        this.StatusCodes = data.StatusCodes;
        this.Status = data.Status;
    }

    toString() {
        return `{BestGuessName: ${this.BestGuessName ? this.BestGuessName.toString() : 'null'}\n` +
               `NameIn: ${this.NameIn}\n` +
               `NameClassification: ${this.NameClassification}\n` +
               `Prefix: ${this.Prefix}\n` +
               `FirstName: ${this.FirstName}\n` +
               `MiddleName: ${this.MiddleName}\n` +
               `LastName: ${this.LastName}\n` +
               `Suffix: ${this.Suffix}\n` +
               `FirstNameFound: ${this.FirstNameFound}\n` +
               `IsCommonFirstName: ${this.IsCommonFirstName}\n` +
               `FirstNameOrigin: ${this.FirstNameOrigin}\n` +
               `FirstNameSimilar: ${this.FirstNameSimilar}\n` +
               `LastNameFound: ${this.LastNameFound}\n` +
               `IsCommonLastName: ${this.IsCommonLastName}\n` +
               `LastNameOrigin: ${this.LastNameOrigin}\n` +
               `LastNameSimilar: ${this.LastNameSimilar}\n` +
               `Gender: ${this.Gender}\n` +
               `FirstNameAlt: ${this.FirstNameAlt}\n` +
               `MiddleNameAlt: ${this.MiddleNameAlt}\n` +
               `LastNameAlt: ${this.LastNameAlt}\n` +
               `FirstNameAltFound: ${this.FirstNameAltFound}\n` +
               `LastNameAltFound: ${this.LastNameAltFound}\n` +
               `GenderAlt: ${this.GenderAlt}\n` +
               `RelatedNames: ${this.RelatedNames}\n` +
               `IsCorrectedName: ${this.IsCorrectedName}\n` +
               `IsBusinessName: ${this.IsBusinessName}\n` +
               `BusinessName: ${this.BusinessName}\n` +
               `VulgarityScore: ${this.VulgarityScore}\n` +
               `CelebrityScore: ${this.CelebrityScore}\n` +
               `BogusScore: ${this.BogusScore}\n` +
               `GarbageScore: ${this.GarbageScore}\n` +
               `FirstNameDictionaryScore: ${this.FirstNameDictionaryScore}\n` +
               `MiddleNameDictionaryScore: ${this.MiddleNameDictionaryScore}\n` +
               `LastNameDictionaryScore: ${this.LastNameDictionaryScore}\n` +
               `OverallNameScore: ${this.OverallNameScore}\n` +
               `IsNameGood: ${this.IsNameGood}\n` +
               `StatusCodes: ${this.StatusCodes}\n` +
               `Status: ${this.Status}\n`;
    }
}

class Error {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
    }

    toString() {
        return `Type: ${this.Type}\n` +
               `TypeCode: ${this.TypeCode}\n` +
               `Desc: ${this.Desc}\n` +
               `DescCode: ${this.DescCode} `;
    }
}

class NameInfoV2Response {
    constructor(data = {}) {
        this.NameInfoV2 = data.NameInfoV2 ? new NameInfoV2(data.NameInfoV2) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `NameInfoV2: ${(this.NameInfoV2 ? this.NameInfoV2.toString() : 'null')}\n` +
               `Error: ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

export{ NameInfoV2Response };